### PR TITLE
Taef Registry Usage Fix

### PR DESCRIPTION
**Pull request by cltshivash**
- Removed the usage of taef registry
- Minor changes to consolidate the version checks

### CODE REVIEW

File: Tasks/VsTest/vstest.ts [hunk 1]
```diff
@@ -963,4 +939,59 @@ function isNullOrWhitespace(input) {
         return true;
     }
     return input.replace(/\s/g, '').length < 1;
-}
\ No newline at end of file
+}
+
+class VSTestVersion {
```
Comment (nigurr) [inline Tasks/VsTest/vstest.ts]: >VSTestVersion [](start = 6, length = 13)

VsTestVersion should be an interface.

Typescript supports anonymous interface implementation. Dev14/Dev15 these shouldn't be classes. they should be anonymous implementations

**Comment by nigurr** [comment]
:clock1:

File: Tasks/VsTest/vstest.ts [hunk 2]
```diff
@@ -172,6 +183,15 @@ function getVstestArguments(settingsFile: string, tiaEnabled: boolean): string[]
     return argsArray;
 }
 
+function isDebugEnabled(): boolean {
```
Comment (nigurr) [inline Tasks/VsTest/vstest.ts]: >isDebugEnabled [](start = 9, length = 14)

I am not sure why are we pushing /diag if system.debug=true

If these diag files are not published, it's not useful

File: Tasks/VsTest/versionfinder.ts [hunk 3]
```diff
@@ -7,28 +7,28 @@ import utils = require('./helpers');
...
-        then( function(exeInfo) {
-            var vstestConsoleInfo = exeInfo;
-            if(exeInfo){
-                vstestConsoleInfo.location = path.join(exeInfo.location, "vstest.console.exe");
-            }            
-            deferred.resolve(vstestConsoleInfo);
-        });
+export async function locateVSTestConsole(testConfig): Promise<string> {
+    let deferred = Q.defer<string>();
+    let vstestExeFolder = await locateTestWindow(testConfig);
```
Comment (nigurr) [inline Tasks/VsTest/versionfinder.ts]: no need to create defer instance. return directly Q.resolve

Comment (nigurr) [inline Tasks/VsTest/versionfinder.ts]: same here.

File: Tasks/VsTest/versionfinder.ts [hunk 4]
```diff
@@ -7,28 +7,28 @@ import utils = require('./helpers');
...
+        vstestExePath = path.join(vstestExeFolder, "vstest.console.exe");
+    }
+
+    deferred.resolve(vstestExePath);
     return deferred.promise;
 }
 
-export function locateTestWindow(testConfig: models.TestConfigurations): Q.Promise<models.ExecutabaleInfo> {
-    let deferred = Q.defer<models.ExecutabaleInfo>();
+export function locateTestWindow(testConfig: models.TestConfigurations): Q.Promise<string> {
```
Comment (nigurr) [inline Tasks/VsTest/versionfinder.ts]: there's fundamental difference between Q.Promise vs Promise. 

Q is external npm lib
Promise is ES6 standard lib and always use Promise and get rid of Q.Promise

**Comment by nigurr** [comment]
>     if(vsVersion === 15) {

it should be number

---
Refers to: Tasks/VsTest/settingshelper.ts:298 in 027490a. [](commit_id = 027490a1574e8a3485cec5c9e67a6b5ba6d99746, deletion_comment = False)

File: Tasks/VsTest/vstest.ts (hunk 1 above)
Comment (cltshivash) [inline Tasks/VsTest/vstest.ts]: Unless there's a perf penalty assocaiated, i would prefer this model for readability.

File: Tasks/VsTest/vstest.ts (hunk 2 above)
Comment (cltshivash) [inline Tasks/VsTest/vstest.ts]: Fair feedback.. This is existing code (related to TIA) will check with Siddhartha.

File: Tasks/VsTest/versionfinder.ts (hunk 4 above)
Comment (cltshivash) [inline Tasks/VsTest/versionfinder.ts]: Done.

File: Tasks/VsTest/vstest.ts [hunk 5]
```diff
@@ -963,4 +939,59 @@ function isNullOrWhitespace(input) {
...
+    }
+
+    isTestImpactSupported(): boolean {
+        return (this.patchNumber >= 25727);
+    }
+
+    vstestDiagSupported(): boolean {
+        return (this.patchNumber > 25428);
+    }
+}
```
Comment (acesiddhu) [inline Tasks/VsTest/vstest.ts]: should we move this to different file. this file is any ways too big

File: Tasks/VsTest/vstest.ts (hunk 1 above)
Comment (nigurr) [inline Tasks/VsTest/vstest.ts]: the things with approach is the the classes will keep on getting added for every version you need to support something.
Ideally a simple function method in js should resolve that and create anonymously. Simple and elegant.

**Pull request merged**